.github/workflows/workflow.yml

```diff
@@ -8,6 +8,24 @@ on:
     branches: [master]
 
 jobs:
+  pre-commit-hooks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    steps:
+      - name: Checkout Code Repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      # Run all pre-commit hooks on all the files.
+      # Getting only staged files can be tricky in case a new PR is opened
+      # since the action is run on a branch in detached head state
```

Comment:
In order to only run `pre-commit` on staged files, the CI workflow would need to detach from `HEAD`, which is the commit that triggered the build. 

I think it would need to do something like `git reset --soft HEAD~` to rollback to the previous commit and keep the diffs. The logic to implement this might make the process convoluted and add additional complexity.

I can't find the source from where I copied the comment from, but it was added because the local `pre-commit` only checks staged files.

Reply:
In order to only run `pre-commit` on staged files, the CI workflow would need to detach from `HEAD`, which is the commit that triggered the build. 

I think we would need to run something like `git reset --soft HEAD~` to rollback to the previous commit and keep the diffs before running `pre-commit`. The logic to implement this might make the process convoluted and add additional complexity.

I can't find the source from where I copied the comment from, but it was added because the local `pre-commit` only checks staged files.